editor/src/core/model/project-file-utils.ts

```diff
@@ -178,6 +180,47 @@ export function getOrDefaultScenes(parsedSuccess: ParseSuccess): UtopiaJSXCompon
   return defaultEmptyUtopiaComponent
 }
 
+export function getSceneElementsFromParseSuccess(success: ParseSuccess): JSXElement[] {
+  const rootElement = getOrDefaultScenes(success).rootElement
+  if (!isJSXElement(rootElement) || rootElement.name.baseVariable !== 'Storyboard') {
+    throw new Error('the root element must be a Storyboard component')
+  }
+  return rootElement.children.filter(
+    (child): child is JSXElement => isJSXElement(child) && isSceneElement(child),
+  )
+}
+
+export function applyUtopiaJSXComponentsChanges(
+  topLevelElements: Array<TopLevelElement>,
+  newUtopiaComponents: Array<UtopiaJSXComponent>,
+): Array<TopLevelElement> {
+  // Run through the old top level elements, replacing the exported elements with those in the
+  // newly updated result with the same name.
+  // If it doesn't exist in the updated result, delete it.
+  // For any new items in the updated result, add them in.
+  const addedSoFar: Set<string> = Utils.emptySet()
+  let newTopLevelElements: Array<TopLevelElement> = []
+  Utils.fastForEach(topLevelElements, (oldTopLevelElement) => {
+    if (isUtopiaJSXComponent(oldTopLevelElement)) {
+      const updatedElement = newUtopiaComponents.find((e) => e.name === oldTopLevelElement.name)
```

Comment:
Super minor, but might help a little with clarity.
```suggestion
      const updatedComponent = newUtopiaComponents.find((e) => e.name === oldTopLevelElement.name)
```